src/marks/vector.d.ts

```diff
@@ -3,29 +3,78 @@ import type {Data, FrameAnchor, MarkOptions, RenderableMark} from "../mark.js";
 
 export type VectorShapeName = "arrow" | "spike";
 
+/**
+ * A custom vector shape implementation.
+ */
 export interface VectorShapeImplementation {
   draw(context: CanvasPath, length: number, radius: number): void;
 }
 
 export type VectorShapeSpec = VectorShapeName | VectorShapeImplementation;
 
 export interface VectorOptions extends MarkOptions {
+  /** A channel for the horizontal position. */
```

Comment:
Perhaps obvious, but we should mention which scale (here *x*) a channel is bound to by default.